File: Easy/Detective Geek/Detective Geek.cs
using System;

class Solution
{
    static void Main(string[] args)
    {
        var time = Convert.ToString(Convert.ToInt32(Console.ReadLine().Replace('#', '1').Replace('*','0'),2)).PadLeft(4,'0').Insert(2,":");
        var address = Console.ReadLine().Split(' ');
        string[] months = new string[] {"jan", "feb", "mar", "apr", "may", "jun", "jul", "aug", "sep", "oct", "nov", "dec"};

        for (int i = 0; i < address.Length; i++)
            address[i] = Convert.ToString(Convert.ToChar(Array.IndexOf(months, address[i].Substring(0,3))*12
                       +Array.IndexOf(months, address[i].Substring(3))));

        Console.WriteLine($"{time}\n{String.Join(String.Empty,address)}");
    }
}

File: Easy/Detective Geek/Detective Geek.py
time = str(int(input().replace('#', '1').replace('*','0'), 2)).zfill(4)
time = time[0:2]+":"+time[2:]
address = input().split(" ")
months = ("jan", "feb", "mar", "apr", "may", "jun", "jul", "aug", "sep", "oct", "nov", "dec")
answer = ""

for i in range(len(address)):
    answer += chr(months.index(address[i][0:3])*12+months.index(address[i][3:]))

print(time)
print(answer)